android/java/org/chromium/chrome/browser/firstrun/WelcomeOnboardingActivity.java

```diff
@@ -0,0 +1,335 @@
+import org.chromium.chrome.browser.set_default_browser.BraveSetDefaultBrowserUtils;
+import org.chromium.chrome.browser.util.BraveConstants;
+import org.chromium.chrome.browser.util.PackageUtils;
+import org.chromium.ui.base.DeviceFormFactor;
+
+import java.lang.Math;
+
+public class WelcomeOnboardingActivity extends FirstRunActivityBase {
+    // mInitializeViewsDone and mInvokePostWorkAtInitializeViews are accessed
+    // from the same thread, so no need to use extra locks
+    private static final String P3A_URL = "https://brave.com/privacy/browser/#how-we-improve-brave";
+    private boolean mInitializeViewsDone;
+    private boolean mInvokePostWorkAtInitializeViews;
+    private boolean mIsP3aEnabled;
+    private boolean isTablet;
+    private FirstRunFlowSequencer mFirstRunFlowSequencer;
+    private int mCurrentStep = -1;
+
+    private View mVLeafAlignTop;
+    private View mVLeafAlignBottom;
+    private ImageView mIvBackground;
+    private ImageView mIvLeafTop;
+    private ImageView mIvLeafBottom;
+    private ImageView mIvBrave;
+    private ImageView mIvArrowDown;
+    private LinearLayout mLayoutCard;
+    private LinearLayout mLayoutCrash;
+    private LinearLayout mLayoutP3a;
+    private TextView mTvWelcome;
+    private TextView mTvCard;
+    private TextView mTvDefault;
+    private Button mBtnPositive;
+    private Button mBtnNegative;
+    private CheckBox mCheckboxCrash;
+    private CheckBox mCheckboxP3a;
+
+    private void initializeViews() {
+        assert !mInitializeViewsDone;
+        setContentView(R.layout.activity_welcome_onboarding);
+
+        isTablet = DeviceFormFactor.isNonMultiDisplayContextOnTablet(this);
+
+        initViews();
+        onClickViews();
+
+        mInitializeViewsDone = true;
+        if (mInvokePostWorkAtInitializeViews) {
+            finishNativeInitializationPostWork();
+        }
+    }
+
+    private void initViews() {
+        mIvBackground = findViewById(R.id.iv_background);
+        mIvLeafTop = findViewById(R.id.iv_leaf_top);
+        mIvLeafBottom = findViewById(R.id.iv_leaf_bottom);
+        mVLeafAlignTop = findViewById(R.id.view_leaf_top_align);
+        mVLeafAlignBottom = findViewById(R.id.view_leaf_bottom_align);
+        mIvBrave = findViewById(R.id.iv_brave);
+        mIvArrowDown = findViewById(R.id.iv_arrow_down);
+        mLayoutCard = findViewById(R.id.layout_card);
+        mLayoutCrash = findViewById(R.id.layout_crash);
+        mLayoutP3a = findViewById(R.id.layout_p3a);
+        mTvWelcome = findViewById(R.id.tv_welcome);
+        mTvCard = findViewById(R.id.tv_card);
+        mTvDefault = findViewById(R.id.tv_default);
+        mCheckboxCrash = findViewById(R.id.checkbox_crash);
+        mCheckboxP3a = findViewById(R.id.checkbox_p3a);
+        mBtnPositive = findViewById(R.id.btn_positive);
+        mBtnNegative = findViewById(R.id.btn_negative);
+
+        int margin;
+        if (isTablet) {
```

Comment:
@tapanmodh nit could be replaced with 
`int margin = isTablet ? 200 : 50;`

Reply:
right, will do it.